app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/ElementEditsController.kt

```diff
@@ -69,18 +73,27 @@ class ElementEditsController(
     }
 
     fun markSynced(edit: ElementEdit, elementUpdates: MapDataUpdates) {
+        val idUpdatesMap = elementUpdates.idUpdates.associate {
+            ElementKey(it.elementType, it.oldElementId) to it.newElementId
+        }
         val syncSuccess: Boolean
         synchronized(this) {
-            for (update in elementUpdates.idUpdates) {
-                editsDB.updateElementId(update.elementType, update.oldElementId, update.newElementId)
+            val editIdsToUpdate = HashSet<Long>()
+            elementUpdates.idUpdates.flatMapTo(editIdsToUpdate) {
+                editElementsDB.getAllByElement(it.elementType, it.oldElementId)
+            }
+            for (id in editIdsToUpdate) {
+                val oldEdit = editsDB.get(id) ?: continue
+                val updatedEdit = oldEdit.copy(action = oldEdit.action.idsUpdatesApplied(idUpdatesMap))
+                editsDB.put(updatedEdit)
+                // must clear first because the element ids associated with this id are different now
+                editElementsDB.delete(id)
+                editElementsDB.put(id, updatedEdit.action.elementKeys)
```

Comment:
Is there a reason why `oldEdit` is not deleted?
It shouldn't matter much because the upload will fail anyway, but it "doesn't feel right"

Reply:
some more possible issues (or just me not understanding what's going on?):
* should `editElementsDB.put(id, updatedEdit.action.elementKeys)` rather use `updatedEdit.id`?
* should `elementIdProviderDB` also be updated to use the new edit id?

Reply:
1. the `oldEdit` is supposed to be overwritten with `editsDB.put(updatedEdit)`. The `id` is the primary key. Will add a test for that.